ESP32 Build is broken, Does not enumerate

### Operating System

Linux

### INFO_UF2.TXT

TinyUF2 Bootloader 0.11.0-13-g9796a54 - tinyusb (0.12.0-203-ga4cfd1c69)
Model: Unexpected Maker TinyS3
Board-ID: ESP32S3-TinyS3-01
Date: Nov  8 2022


### What happened ?

ESP32 build worked fine until mid november, since then it is broken, it does not enumerate. Maybe it is cause by updated docker image for esp-idf.  The image used by the workflow is repressive/idf:release-v4.4 which may have been updated at that time... last updated 5 days ago!

### How to reproduce ?

I am testing on unexpectedmaker tinys3 board! Here are the last working and first non-working builds that I found!

(Nov 9) unexpectedmaker_tiny3  boots fine
https://github.com/adafruit/tinyuf2/actions/runs/3424066037

(Nov 15) couple of commits later it does not enumerate
https://github.com/adafruit/tinyuf2/actions/runs/3466891115

Can anyone try to reproduce the issue? What could be wrong?

### Debug Log

_No response_

### Screenshots

_No response_

### ESP32-S3-DevKitC1-N32R8V (WROOM-2)
### Setup
My next approach was to compile from source. I started by defining a new board with the associated files. In this instance, I defined it as `espressif_esp32s3_devkitc_1-1` and under it, I included the following files.

**partitions-32MB.csv**
```
# ESP-IDF Partition Table
# Name,   Type, SubType, Offset,  Size, Flags
# bootloader.bin,,          0x1000, 32K
# partition table,,         0x8000, 4K
nvs,      data, nvs,      0x9000,  20K,
otadata,  data, ota,      0xe000,  8K,
ota_0,    0,    ota_0,   0x10000,  2048K,
ota_1,    0,    ota_1,  0x210000,  2048K,
uf2,      app,  factory,0x410000,  256K,
ffat,     data, fat,    0x450000,  28352K,
```

**sdkconfig**
```
# Board Specific Config

# Partition Table
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-32MB.csv"

# Serial flasher config
CONFIG_ESPTOOLPY_FLASHSIZE_32MB=y
```

**board.h**
I have seen a few comments that on the WROOM-2, the Neopixel pin is `GPIO38` instead of `GPIO48`.
```
# Apply board specific content here
set(IDF_TARGET "esp32s3")
/* 
 * The MIT License (MIT)
 *
 * Copyright (c) 2020 Ha Thach (tinyusb.org) for Adafruit Industries
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

#ifndef ESPRESSIF_S3_DEVKITC_H_
#define ESPRESSIF_S3_DEVKITC_H_

//--------------------------------------------------------------------+
// Button
//--------------------------------------------------------------------+

// Enter UF2 mode if GPIO is pressed while 2nd stage bootloader indicator
// is on e.g RGB = Purple. If it is GPIO0, user should not hold this while
// reset since that will instead run the 1st stage ROM bootloader
#define PIN_BUTTON_UF2        0

// GPIO that implement 1-bit memory with RC components which hold the
// pin value long enough for double reset detection.
// #define PIN_DOUBLE_RESET_RC

//--------------------------------------------------------------------+
// LED
//--------------------------------------------------------------------+

// GPIO connected to Neopixel data
#define NEOPIXEL_PIN          38

// Brightness percentage from 1 to 255
#define NEOPIXEL_BRIGHTNESS   0x10

// Number of neopixels
#define NEOPIXEL_NUMBER       1


// LED for indicator
// If not defined neopixel will be use for flash writing instead
// #define LED_PIN               42
// #define LED_STATE_ON          1

//--------------------------------------------------------------------+
// USB UF2
//--------------------------------------------------------------------+

#define USB_VID           0x239A
#define USB_PID           0x00A5    // TODO temporarily shared with S2 saola wrover
#define USB_MANUFACTURER  "Espressif"
#define USB_PRODUCT       "ESP32S3 DevKitC 1"

#define UF2_PRODUCT_NAME  USB_MANUFACTURER " " USB_PRODUCT
#define UF2_BOARD_ID      "ESP32S3-DevKitC-v1.1"
#define UF2_VOLUME_LABEL  "S3DKC1BOOT"
#define UF2_INDEX_URL     "https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html"

#endif
# Board Specific Config

# Partition Table
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-32MB.csv"

# Serial flasher config
CONFIG_ESPTOOLPY_FLASHSIZE_32MB=y
```

### Compile
When I tried to compile I ran into issues generating the partitions tables. `esptool.py` (v3.2-dev) does not support flash sizes higher than 16.

`make BOARD=espressif_esp32s3_devkitc_1-1 all`
<details><summary>Show output</summary>

```
idf.py -B_build/espressif_esp32s3_devkitc_1-1 -DBOARD=espressif_esp32s3_devkitc_1-1 build
Executing action: all (aliases: build)
Running ninja in directory /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1
Executing "ninja all"...
[0/1] Re-running CMake...-- Building ESP-IDF components for target esp32s3
-- Project sdkconfig file /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/sdkconfig
Loading defaults file /home/rgrizzell/tinyuf2/ports/espressif/sdkconfig.defaults...
/home/rgrizzell/tinyuf2/ports/espressif/sdkconfig.defaults:21 CONFIG_FLASHMODE_QIO was replaced with CONFIG_ESPTOOLPY_FLASHMODE_QIO
Loading defaults file /home/rgrizzell/tinyuf2/ports/espressif/boards/espressif_esp32s3_devkitc_1-1/sdkconfig...
-- Found Python3: /home/rwgrizzell/.espressif/python_env/idf5.0_py3.10_env/bin/python (found version "3.10.6") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "tinyuf2" version: 0.11.0-dirty
-- Adding linker script /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_system/ld/esp32s3/sections.ld.in
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.api.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.libgcc.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.version.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/soc/esp32s3/ld/esp32s3.peripherals.ld
-- Components: app_trace app_update boards bootloader bootloader_support bt cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_netif esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json lcd led_strip log lwip mbedtls mqtt newlib nvs_flash openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs src tcp_transport touch_element ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: /home/rgrizzell/tinyuf2/lib/esp-idf/components/app_trace /home/rgrizzell/tinyuf2/lib/esp-idf/components/app_update /home/rgrizzell/tinyuf2/ports/espressif/boards /home/rgrizzell/tinyuf2/ports/espressif/components/bootloader /home/rgrizzell/tinyuf2/lib/esp-idf/components/bootloader_support /home/rgrizzell/tinyuf2/lib/esp-idf/components/bt /home/rgrizzell/tinyuf2/lib/esp-idf/components/cmock /home/rgrizzell/tinyuf2/lib/esp-idf/components/console /home/rgrizzell/tinyuf2/lib/esp-idf/components/cxx /home/rgrizzell/tinyuf2/lib/esp-idf/components/driver /home/rgrizzell/tinyuf2/lib/esp-idf/components/efuse /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp-tls /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_adc /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_app_format /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_common /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_eth /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_event /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_gdbstub /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_hid /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_http_client /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_http_server /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_https_ota /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_https_server /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_hw_support /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_lcd /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_local_ctrl /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_netif /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_partition /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_phy /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_pm /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_psram /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_ringbuf /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_system /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_timer /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_wifi /home/rgrizzell/tinyuf2/lib/esp-idf/components/espcoredump /home/rgrizzell/tinyuf2/lib/esp-idf/components/esptool_py /home/rgrizzell/tinyuf2/lib/esp-idf/components/fatfs /home/rgrizzell/tinyuf2/lib/esp-idf/components/freertos /home/rgrizzell/tinyuf2/lib/esp-idf/components/hal /home/rgrizzell/tinyuf2/lib/esp-idf/components/heap /home/rgrizzell/tinyuf2/lib/esp-idf/components/http_parser /home/rgrizzell/tinyuf2/lib/esp-idf/components/idf_test /home/rgrizzell/tinyuf2/lib/esp-idf/components/ieee802154 /home/rgrizzell/tinyuf2/lib/esp-idf/components/json /home/rgrizzell/tinyuf2/ports/espressif/components/lcd /home/rgrizzell/tinyuf2/ports/espressif/components/led_strip /home/rgrizzell/tinyuf2/lib/esp-idf/components/log /home/rgrizzell/tinyuf2/lib/esp-idf/components/lwip /home/rgrizzell/tinyuf2/lib/esp-idf/components/mbedtls /home/rgrizzell/tinyuf2/lib/esp-idf/components/mqtt /home/rgrizzell/tinyuf2/lib/esp-idf/components/newlib /home/rgrizzell/tinyuf2/lib/esp-idf/components/nvs_flash /home/rgrizzell/tinyuf2/lib/esp-idf/components/openthread /home/rgrizzell/tinyuf2/lib/esp-idf/components/partition_table /home/rgrizzell/tinyuf2/lib/esp-idf/components/perfmon /home/rgrizzell/tinyuf2/lib/esp-idf/components/protobuf-c /home/rgrizzell/tinyuf2/lib/esp-idf/components/protocomm /home/rgrizzell/tinyuf2/lib/esp-idf/components/pthread /home/rgrizzell/tinyuf2/lib/esp-idf/components/sdmmc /home/rgrizzell/tinyuf2/lib/esp-idf/components/soc /home/rgrizzell/tinyuf2/lib/esp-idf/components/spi_flash /home/rgrizzell/tinyuf2/lib/esp-idf/components/spiffs /home/rgrizzell/tinyuf2/src /home/rgrizzell/tinyuf2/lib/esp-idf/components/tcp_transport /home/rgrizzell/tinyuf2/lib/esp-idf/components/touch_element /home/rgrizzell/tinyuf2/lib/esp-idf/components/ulp /home/rgrizzell/tinyuf2/lib/esp-idf/components/unity /home/rgrizzell/tinyuf2/lib/esp-idf/components/usb /home/rgrizzell/tinyuf2/lib/esp-idf/components/vfs /home/rgrizzell/tinyuf2/lib/esp-idf/components/wear_levelling /home/rgrizzell/tinyuf2/lib/esp-idf/components/wifi_provisioning /home/rgrizzell/tinyuf2/lib/esp-idf/components/wpa_supplicant /home/rgrizzell/tinyuf2/lib/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1
[4/917] Generating ../../partition_table/partition-table.binFAILED: partition_table/partition-table.bin /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/partition_table/partition-table.bin 
cd /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/esp-idf/partition_table && /home/rwgrizzell/.espressif/python_env/idf5.0_py3.10_env/bin/python /home/rgrizzell/tinyuf2/lib/esp-idf/components/partition_table/gen_esp32part.py -q --offset 0x8000 --flash-size 2MB -- /home/rgrizzell/tinyuf2/ports/espressif/partitions-32MB.csv /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/partition_table/partition-table.bin && /usr/bin/cmake -E echo "Partition table binary generated. Contents:" && /usr/bin/cmake -E echo "*******************************************************************************" && /home/rwgrizzell/.espressif/python_env/idf5.0_py3.10_env/bin/python /home/rgrizzell/tinyuf2/lib/esp-idf/components/partition_table/gen_esp32part.py -q --offset 0x8000 --flash-size 2MB -- /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/partition_table/partition-table.bin && /usr/bin/cmake -E echo "*******************************************************************************"
Partitions tables occupies 32.0MB of flash (33554432 bytes) which does not fit in configured flash size 2MB. Change the flash size in menuconfig under the 'Serial Flasher Config' menu.
[37/917] Building C object esp-idf/mbedtls/mbedtls/library/CMakeFiles/mbedcrypto.dir/bignum.c.objninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/log/idf_py_stderr_output_748177 and /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/log/idf_py_stdout_output_748177
make: *** [Makefile:31: all] Error 2
```
</details>


I then updated the `esp-idf` submodule to v5.0 (which updates esptool.py to v4.4), the Serial Flasher then supports 32MB, recognizing the variables set in `sdkconfig`. Confirmed by running: `idf.py menuconfig -DBOARD=espressif_esp32s3_devkitc_1-1`

<details><summary>Show output</summary>

```
idf.py -B_build/espressif_esp32s3_devkitc_1-1 -DBOARD=espressif_esp32s3_devkitc_1-1 build
Executing action: all (aliases: build)
Running cmake in directory /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1
Executing "cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -DBOARD=espressif_esp32s3_devkitc_1-1 -DCCACHE_ENABLE=0 /home/rgrizzell/tinyuf2/ports/espressif"...
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- The C compiler identification is GNU 11.2.0
-- The CXX compiler identification is GNU 11.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /home/rgrizzell/.espressif/tools/xtensa-esp32s3-elf/esp-2022r1-11.2.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /home/rgrizzell/.espressif/tools/xtensa-esp32s3-elf/esp-2022r1-11.2.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /home/rgrizzell/.espressif/tools/xtensa-esp32s3-elf/esp-2022r1-11.2.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32s3
-- Project sdkconfig file /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/sdkconfig
Loading defaults file /home/rgrizzell/tinyuf2/ports/espressif/sdkconfig.defaults...
/home/rgrizzell/tinyuf2/ports/espressif/sdkconfig.defaults:21 CONFIG_FLASHMODE_QIO was replaced with CONFIG_ESPTOOLPY_FLASHMODE_QIO
Loading defaults file /home/rgrizzell/tinyuf2/ports/espressif/boards/espressif_esp32s3_devkitc_1-1/sdkconfig...
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Found Python3: /home/rgrizzell/.espressif/python_env/idf5.0_py3.10_env/bin/python (found version "3.10.6") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "tinyuf2" version: 0.11.0-dirty
-- Adding linker script /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/esp-idf/esp_system/ld/memory.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_system/ld/esp32s3/sections.ld.in
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.api.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.libgcc.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.newlib.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom/esp32s3/ld/esp32s3.rom.version.ld
-- Adding linker script /home/rgrizzell/tinyuf2/lib/esp-idf/components/soc/esp32s3/ld/esp32s3.peripherals.ld
-- Components: app_trace app_update boards bootloader bootloader_support bt cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_netif esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json lcd led_strip log lwip mbedtls mqtt newlib nvs_flash openthread partition_table perfmon protobuf-c protocomm pthread sdmmc soc spi_flash spiffs src tcp_transport touch_element ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant xtensa
-- Component paths: /home/rgrizzell/tinyuf2/lib/esp-idf/components/app_trace /home/rgrizzell/tinyuf2/lib/esp-idf/components/app_update /home/rgrizzell/tinyuf2/ports/espressif/boards /home/rgrizzell/tinyuf2/ports/espressif/components/bootloader /home/rgrizzell/tinyuf2/lib/esp-idf/components/bootloader_support /home/rgrizzell/tinyuf2/lib/esp-idf/components/bt /home/rgrizzell/tinyuf2/lib/esp-idf/components/cmock /home/rgrizzell/tinyuf2/lib/esp-idf/components/console /home/rgrizzell/tinyuf2/lib/esp-idf/components/cxx /home/rgrizzell/tinyuf2/lib/esp-idf/components/driver /home/rgrizzell/tinyuf2/lib/esp-idf/components/efuse /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp-tls /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_adc /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_app_format /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_common /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_eth /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_event /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_gdbstub /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_hid /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_http_client /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_http_server /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_https_ota /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_https_server /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_hw_support /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_lcd /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_local_ctrl /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_netif /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_partition /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_phy /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_pm /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_psram /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_ringbuf /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_rom /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_system /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_timer /home/rgrizzell/tinyuf2/lib/esp-idf/components/esp_wifi /home/rgrizzell/tinyuf2/lib/esp-idf/components/espcoredump /home/rgrizzell/tinyuf2/lib/esp-idf/components/esptool_py /home/rgrizzell/tinyuf2/lib/esp-idf/components/fatfs /home/rgrizzell/tinyuf2/lib/esp-idf/components/freertos /home/rgrizzell/tinyuf2/lib/esp-idf/components/hal /home/rgrizzell/tinyuf2/lib/esp-idf/components/heap /home/rgrizzell/tinyuf2/lib/esp-idf/components/http_parser /home/rgrizzell/tinyuf2/lib/esp-idf/components/idf_test /home/rgrizzell/tinyuf2/lib/esp-idf/components/ieee802154 /home/rgrizzell/tinyuf2/lib/esp-idf/components/json /home/rgrizzell/tinyuf2/ports/espressif/components/lcd /home/rgrizzell/tinyuf2/ports/espressif/components/led_strip /home/rgrizzell/tinyuf2/lib/esp-idf/components/log /home/rgrizzell/tinyuf2/lib/esp-idf/components/lwip /home/rgrizzell/tinyuf2/lib/esp-idf/components/mbedtls /home/rgrizzell/tinyuf2/lib/esp-idf/components/mqtt /home/rgrizzell/tinyuf2/lib/esp-idf/components/newlib /home/rgrizzell/tinyuf2/lib/esp-idf/components/nvs_flash /home/rgrizzell/tinyuf2/lib/esp-idf/components/openthread /home/rgrizzell/tinyuf2/lib/esp-idf/components/partition_table /home/rgrizzell/tinyuf2/lib/esp-idf/components/perfmon /home/rgrizzell/tinyuf2/lib/esp-idf/components/protobuf-c /home/rgrizzell/tinyuf2/lib/esp-idf/components/protocomm /home/rgrizzell/tinyuf2/lib/esp-idf/components/pthread /home/rgrizzell/tinyuf2/lib/esp-idf/components/sdmmc /home/rgrizzell/tinyuf2/lib/esp-idf/components/soc /home/rgrizzell/tinyuf2/lib/esp-idf/components/spi_flash /home/rgrizzell/tinyuf2/lib/esp-idf/components/spiffs /home/rgrizzell/tinyuf2/src /home/rgrizzell/tinyuf2/lib/esp-idf/components/tcp_transport /home/rgrizzell/tinyuf2/lib/esp-idf/components/touch_element /home/rgrizzell/tinyuf2/lib/esp-idf/components/ulp /home/rgrizzell/tinyuf2/lib/esp-idf/components/unity /home/rgrizzell/tinyuf2/lib/esp-idf/components/usb /home/rgrizzell/tinyuf2/lib/esp-idf/components/vfs /home/rgrizzell/tinyuf2/lib/esp-idf/components/wear_levelling /home/rgrizzell/tinyuf2/lib/esp-idf/components/wifi_provisioning /home/rgrizzell/tinyuf2/lib/esp-idf/components/wpa_supplicant /home/rgrizzell/tinyuf2/lib/esp-idf/components/xtensa
-- Configuring done
-- Generating done
-- Build files have been written to: /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1
Running ninja in directory /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1
Executing "ninja all"...
[28/917] Generating ../../partition_table/partition-table.binPartition table binary generated. Contents:
*******************************************************************************
# ESP-IDF Partition Table
# Name, Type, SubType, Offset, Size, Flags
nvs,data,nvs,0x9000,20K,
otadata,data,ota,0xe000,8K,
ota_0,app,ota_0,0x10000,2M,
ota_1,app,ota_1,0x210000,2M,
uf2,app,factory,0x410000,256K,
ffat,data,fat,0x450000,28352K,
*******************************************************************************
[398/917] Performing configure step for 'bootloader'FAILED: bootloader-prefix/src/bootloader-stamp/bootloader-configure /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/bootloader-prefix/src/bootloader-stamp/bootloader-configure 
cd /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/bootloader && /usr/bin/cmake -DSDKCONFIG=/home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/sdkconfig -DIDF_PATH=/home/rgrizzell/tinyuf2/lib/esp-idf -DIDF_TARGET=esp32s3 -DPYTHON_DEPS_CHECKED=1 -DPYTHON=/home/rgrizzell/.espressif/python_env/idf5.0_py3.10_env/bin/python -DEXTRA_COMPONENT_DIRS=/home/rgrizzell/tinyuf2/ports/espressif/components/bootloader -DPROJECT_SOURCE_DIR=/home/rgrizzell/tinyuf2/ports/espressif -DBOARD=espressif_esp32s3_devkitc_1-1 -DLEGACY_INCLUDE_COMMON_HEADERS= -GNinja /home/rgrizzell/tinyuf2/ports/espressif/components/bootloader/subproject && /usr/bin/cmake -E touch /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/bootloader-prefix/src/bootloader-stamp/bootloader-configure
-- Found Git: /usr/bin/git (found version "2.34.1") 
CMake Error at /home/rgrizzell/tinyuf2/lib/esp-idf/tools/cmake/project.cmake:243 (message):
  Directory specified in EXTRA_COMPONENT_DIRS doesn't exist:
  /home/rgrizzell/tinyuf2/ports/espressif/bootloader_components
Call Stack (most recent call first):
  /home/rgrizzell/tinyuf2/lib/esp-idf/tools/cmake/project.cmake:327 (__project_init)
  CMakeLists.txt:59 (project)


-- Configuring incomplete, errors occurred!
[431/917] Building CXX object esp-idf/nvs_flash/CMakeFiles/__idf_nvs_flash.dir/src/nvs_storage.cpp.objninja: build stopped: subcommand failed.
ninja failed with exit code 1, output of the command is in the /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/log/idf_py_stderr_output_799291 and /home/rgrizzell/tinyuf2/ports/espressif/_build/espressif_esp32s3_devkitc_1-1/log/idf_py_stdout_output_799291
make: *** [Makefile:31: all] Error 2
```

At this point, it seems there are some breaking changes between ESP-IDF v4.4 and v5.0.

--- a/.github/workflows/build_esp32.yml
+++ b/.github/workflows/build_esp32.yml
@@ -99,7 +99,7 @@ jobs:
       run: echo BIN_PATH=ports/espressif/_bin/${{ matrix.board }} >> $GITHUB_ENV
 
     - name: Build
-      run: docker run --rm -v $PWD:/project -w /project espressif/idf:release-v4.4  /bin/bash -c "git config --global --add safe.directory /project && make -C ports/espressif/ BOARD=${{ matrix.board }} all self-update copy-artifact"
+      run: docker run --rm -v $PWD:/project -w /project espressif/idf:v4.4.3 /bin/bash -c "git config --global --add safe.directory /project && make -C ports/espressif/ BOARD=${{ matrix.board }} all self-update copy-artifact"
 
     - uses: actions/upload-artifact@v3
       with:

